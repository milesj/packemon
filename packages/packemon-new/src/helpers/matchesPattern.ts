import micromatch from 'micromatch';

export function matchesPattern(value: string, pattern: string) {
	const patterns = pattern.split(',');

	return micromatch.isMatch(value, patterns) || patterns.includes(value);
}
