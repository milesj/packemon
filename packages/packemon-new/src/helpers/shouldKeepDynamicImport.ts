import { Platform, Support } from '../types';

// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#browser_compatibility
export function shouldKeepDynamicImport(platform: Platform, support: Support): boolean {
	switch (platform) {
		case 'node':
			// >= v13.2
			// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#browser_compatibility
			return support !== 'legacy';
		case 'browser':
			// >= 2019
			// https://caniuse.com/es6-module-dynamic-import
			return true;
		default:
			// RN does not support code splitting
			// https://github.com/facebook/metro/issues/52
			return false;
	}
}
