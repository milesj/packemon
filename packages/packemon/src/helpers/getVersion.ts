import fs from 'fs';
import { PackageStructure } from '@boost/common';

export function getVersion(): string {
	return (
		JSON.parse(
			// @ts-expect-error Not typed for URL
			fs.readFileSync(new URL('../../package.json', import.meta.url), 'utf8'),
		) as PackageStructure
	).version;
}
