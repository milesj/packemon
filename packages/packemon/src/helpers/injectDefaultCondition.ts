/* eslint-disable no-param-reassign */

import { PackageExportConditions, PackageExports } from '../types';

export function injectDefaultCondition(exportMap: PackageExports) {
	Object.entries(exportMap).forEach(([path, conditions]) => {
		if (typeof conditions !== 'object') {
			return;
		}

		// Order from most to least down-leveled
		if (!conditions.default) {
			for (const key of [
				'browser',
				'react-native',
				'electron',
				'node',
				'require',
			] as PackageExportConditions[]) {
				if (conditions[key]) {
					conditions.default = conditions[key];
					delete conditions[key];
					break;
				}
			}
		}

		const keys = Object.keys(conditions);

		if (keys.length === 1) {
			exportMap[path] =
				keys[0] === 'default' ? conditions.default : conditions[keys[0] as PackageExportConditions];
		} else {
			exportMap[path] = conditions;
		}
	});
}
