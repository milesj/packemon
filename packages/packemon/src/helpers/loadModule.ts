import { createRequire } from 'module';

const modRequire = createRequire(import.meta.url);

export function loadModule(name: string, message: string): unknown {
	try {
		return modRequire(name);
	} catch {
		throw new Error(`${message} Please install with \`yarn add --dev ${name}\`.`);
	}
}
