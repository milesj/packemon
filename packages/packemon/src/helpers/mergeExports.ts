/* eslint-disable no-param-reassign */

import { PackageExportConditions, PackageExportPaths } from '../types';

export function mergeExports(
	prev: PackageExportPaths | string,
	base: PackageExportPaths | string,
	root: boolean,
): PackageExportPaths | undefined {
	const next = typeof base === 'string' ? { default: base } : base;

	if (typeof prev === 'string') {
		return next;
	}

	Object.entries(next).forEach(([origKey, nextValue]) => {
		const key = origKey as PackageExportConditions;
		const prevValue = prev[key];

		if (!prevValue) {
			prev[key] = nextValue;
		} else if (nextValue) {
			prev[key] = mergeExports(prevValue, nextValue, false);
		}
	});

	// Always include a default export for tooling to work correctly
	const keys = Object.keys(prev);

	if (root && !prev.default) {
		for (const key of [
			'react-native',
			'electron',
			'browser',
			'node',
			'import',
			'require',
		] as PackageExportConditions[]) {
			if (prev[key]) {
				if (keys.length === 1) {
					return prev[key] as PackageExportPaths;
				}

				prev.default = prev[key];
				delete prev[key];

				break;
			}
		}
	}

	return prev;
}
