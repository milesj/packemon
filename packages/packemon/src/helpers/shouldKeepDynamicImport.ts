import { Format, Platform } from '../types';

// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#browser_compatibility
export function shouldKeepDynamicImport(platform: Platform, format: Format): boolean {
	if (format === 'umd') {
		return false;
	}

	switch (platform) {
		case 'node':
			// >= v13.2
			// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#browser_compatibility
			return true;
		case 'electron':
		case 'browser':
			// >= 2019
			// https://caniuse.com/es6-module-dynamic-import
			return true;
		case 'native':
			// >= RN 0.72
			// https://metrobundler.dev/docs/module-api/#import-dynamic-import
			return true;
		default:
			return false;
	}
}
