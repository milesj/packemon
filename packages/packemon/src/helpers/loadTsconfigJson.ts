import { Path, type PortablePath } from '@boost/common';
import type { FileSystem } from '../FileSystem';
import type { TSConfigStructure } from '../types';
import { loadModule } from './loadModule';

const CACHE = new Map<Path, TSConfigStructure>();

export function loadTsconfigJson(
	path: PortablePath,
	fs: FileSystem,
): TSConfigStructure | undefined {
	const tsconfigJsonPath = Path.create(path);
	const tsconfig = CACHE.get(tsconfigJsonPath);

	if (tsconfig) {
		return tsconfig;
	}

	if (!tsconfigJsonPath.exists()) {
		return undefined;
	}

	const ts = loadModule(
		'typescript',
		'TypeScript is required for config loading.',
	) as typeof import('typescript');

	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
	const { config, error } = ts.readConfigFile(tsconfigJsonPath.path(), fs.readFile);

	const host = {
		getCanonicalFileName: (fileName: string) => fileName,
		getCurrentDirectory: () => ts.sys.getCurrentDirectory(),
		getNewLine: () => ts.sys.newLine,
	};

	// istanbul ignore next
	if (error) {
		throw new Error(ts.formatDiagnostic(error, host));
	}

	const result = ts.parseJsonConfigFileContent(
		config,
		ts.sys,
		tsconfigJsonPath.parent().path(),
		{},
		tsconfigJsonPath.path(),
	);

	// istanbul ignore next
	if (result.errors.length > 0) {
		throw new Error(ts.formatDiagnostics(result.errors, host));
	}

	CACHE.set(tsconfigJsonPath, result);

	return result;
}
