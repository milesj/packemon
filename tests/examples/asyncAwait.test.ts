import { testExampleOutput } from '../helpers';

describe('Async/await', () => {
	testExampleOutput('async-await.ts');
});
