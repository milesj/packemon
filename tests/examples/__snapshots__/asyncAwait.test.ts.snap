// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async/await transforms example test case 1`] = `
Array [
  "/cjs/index-node-current-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: cjs
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-current-cjs.cjs.map
",
]
`;

exports[`Async/await transforms example test case 2`] = `
Array [
  "/cjs/index-node-experimental-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: cjs
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-experimental-cjs.cjs.map
",
]
`;

exports[`Async/await transforms example test case 3`] = `
Array [
  "/cjs/index-node-legacy-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: cjs
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch (_unused) {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-legacy-cjs.cjs.map
",
]
`;

exports[`Async/await transforms example test case 4`] = `
Array [
  "/cjs/index-node-stable-cjs.cjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: cjs
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-stable-cjs.cjs.map
",
]
`;

exports[`Async/await transforms example test case 5`] = `
Array [
  "/esm/index-browser-current-esm.js",
  "function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

var run = function run() {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: esm
var wait = function wait() {
  var _this = this;

  try {
    return Promise.resolve(new Promise(function (resolve) {
      _newArrowCheck(this, _this);

      setTimeout(resolve, 100);
    }.bind(this)));
  } catch (e) {
    return Promise.reject(e);
  }
};

export { run, wait };
//# sourceMappingURL=index-browser-current-esm.js.map
",
]
`;

exports[`Async/await transforms example test case 6`] = `
Array [
  "/esm/index-browser-experimental-esm.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: esm
async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

export { run, wait };
//# sourceMappingURL=index-browser-experimental-esm.js.map
",
]
`;

exports[`Async/await transforms example test case 7`] = `
Array [
  "/esm/index-browser-legacy-esm.js",
  "function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

var run = function run() {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: esm
var wait = function wait() {
  var _this = this;

  try {
    return Promise.resolve(new Promise(function (resolve) {
      _newArrowCheck(this, _this);

      setTimeout(resolve, 100);
    }.bind(this)));
  } catch (e) {
    return Promise.reject(e);
  }
};

export { run, wait };
//# sourceMappingURL=index-browser-legacy-esm.js.map
",
]
`;

exports[`Async/await transforms example test case 8`] = `
Array [
  "/esm/index-browser-stable-esm.js",
  "function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

var run = function run() {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: esm
var wait = function wait() {
  var _this = this;

  try {
    return Promise.resolve(new Promise(function (resolve) {
      _newArrowCheck(this, _this);

      setTimeout(resolve, 100);
    }.bind(this)));
  } catch (e) {
    return Promise.reject(e);
  }
};

export { run, wait };
//# sourceMappingURL=index-browser-stable-esm.js.map
",
]
`;

exports[`Async/await transforms example test case 9`] = `
Array [
  "/lib/index-browser-current-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: current, Format: lib
'use strict';

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

var run = function run() {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

var wait = function wait() {
  var _this = this;

  try {
    return Promise.resolve(new Promise(function (resolve) {
      _newArrowCheck(this, _this);

      setTimeout(resolve, 100);
    }.bind(this)));
  } catch (e) {
    return Promise.reject(e);
  }
};

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-browser-current-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 10`] = `
Array [
  "/lib/index-browser-experimental-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: experimental, Format: lib
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-browser-experimental-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 11`] = `
Array [
  "/lib/index-browser-legacy-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: legacy, Format: lib
'use strict';

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

var run = function run() {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

var wait = function wait() {
  var _this = this;

  try {
    return Promise.resolve(new Promise(function (resolve) {
      _newArrowCheck(this, _this);

      setTimeout(resolve, 100);
    }.bind(this)));
  } catch (e) {
    return Promise.reject(e);
  }
};

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-browser-legacy-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 12`] = `
Array [
  "/lib/index-browser-stable-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: browser, Support: stable, Format: lib
'use strict';

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

var run = function run() {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

var wait = function wait() {
  var _this = this;

  try {
    return Promise.resolve(new Promise(function (resolve) {
      _newArrowCheck(this, _this);

      setTimeout(resolve, 100);
    }.bind(this)));
  } catch (e) {
    return Promise.reject(e);
  }
};

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-browser-stable-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 13`] = `
Array [
  "/lib/index-native-current-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: current, Format: lib
'use strict';

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

const run = function () {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

const wait = function () {
  try {
    return Promise.resolve(new Promise(resolve => {
      setTimeout(resolve, 100);
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-native-current-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 14`] = `
Array [
  "/lib/index-native-experimental-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: experimental, Format: lib
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-native-experimental-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 15`] = `
Array [
  "/lib/index-native-legacy-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: legacy, Format: lib
'use strict';

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

var run = function run() {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

var wait = function wait() {
  var _this = this;

  try {
    return Promise.resolve(new Promise(function (resolve) {
      _newArrowCheck(this, _this);

      setTimeout(resolve, 100);
    }.bind(this)));
  } catch (e) {
    return Promise.reject(e);
  }
};

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-native-legacy-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 16`] = `
Array [
  "/lib/index-native-stable-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: native, Support: stable, Format: lib
'use strict';

function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

const run = function () {
  try {
    return Promise.resolve(_catch(function () {
      return Promise.resolve(wait()).then(function () {});
    }, function () {
      throw new Error('Fail');
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

const wait = function () {
  try {
    return Promise.resolve(new Promise(resolve => {
      setTimeout(resolve, 100);
    }));
  } catch (e) {
    return Promise.reject(e);
  }
};

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-native-stable-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 17`] = `
Array [
  "/lib/index-node-current-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: lib
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-current-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 18`] = `
Array [
  "/lib/index-node-experimental-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: lib
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-experimental-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 19`] = `
Array [
  "/lib/index-node-legacy-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: lib
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch (_unused) {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-legacy-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 20`] = `
Array [
  "/lib/index-node-stable-lib.js",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: lib
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

exports.run = run;
exports.wait = wait;
//# sourceMappingURL=index-node-stable-lib.js.map
",
]
`;

exports[`Async/await transforms example test case 21`] = `
Array [
  "/mjs/index-node-current-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: mjs
async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

export { run, wait };
//# sourceMappingURL=index-node-current-mjs.mjs.map
",
]
`;

exports[`Async/await transforms example test case 22`] = `
Array [
  "/mjs/index-node-experimental-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: mjs
async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

export { run, wait };
//# sourceMappingURL=index-node-experimental-mjs.mjs.map
",
]
`;

exports[`Async/await transforms example test case 23`] = `
Array [
  "/mjs/index-node-legacy-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: mjs
async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch (_unused) {
    throw new Error('Fail');
  }
}

export { run, wait };
//# sourceMappingURL=index-node-legacy-mjs.mjs.map
",
]
`;

exports[`Async/await transforms example test case 24`] = `
Array [
  "/mjs/index-node-stable-mjs.mjs",
  "// Generated with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: mjs
async function wait() {
  return new Promise(resolve => {
    setTimeout(resolve, 100);
  });
}

async function run() {
  try {
    await wait();
  } catch {
    throw new Error('Fail');
  }
}

export { run, wait };
//# sourceMappingURL=index-node-stable-mjs.mjs.map
",
]
`;

exports[`Async/await transforms example test case 25`] = `
Array [
  "/package.json",
  Object {
    "files": Array [
      "cjs/**/*.{cjs,map}",
      "esm/**/*.{js,map}",
      "lib/**/*.{js,map}",
      "mjs/**/*.{mjs,map}",
      "src/**/*.{ts,tsx,json}",
      "umd/**/*.{js,map}",
    ],
    "name": "examples",
  },
]
`;

exports[`Async/await transforms example test case 26`] = `
Array [
  "/umd/index-browser-current-umd.js",
  "function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

(function (global, factory) {
  if (typeof define === \\"function\\" && define.amd) {
    define([\\"exports\\"], factory);
  } else if (typeof exports !== \\"undefined\\") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.examples = mod.exports;
  }
})(typeof globalThis !== \\"undefined\\" ? globalThis : typeof self !== \\"undefined\\" ? self : this, function (_exports) {
  \\"use strict\\";

  Object.defineProperty(_exports, \\"__esModule\\", {
    value: true
  });
  _exports.wait = _exports.run = void 0;

  function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

  var run = function run() {
    try {
      return Promise.resolve(_catch(function () {
        return Promise.resolve(wait()).then(function () {});
      }, function () {
        throw new Error('Fail');
      }));
    } catch (e) {
      return Promise.reject(e);
    }
  };

  _exports.run = run;

  // Generated with Packemon: https://packemon.dev
  // Platform: browser, Support: current, Format: umd
  var wait = function wait() {
    var _this = this;

    try {
      return Promise.resolve(new Promise(function (resolve) {
        _newArrowCheck(this, _this);

        setTimeout(resolve, 100);
      }.bind(this)));
    } catch (e) {
      return Promise.reject(e);
    }
  };

  _exports.wait = wait;
});
//# sourceMappingURL=index-browser-current-umd.js.map
",
]
`;

exports[`Async/await transforms example test case 27`] = `
Array [
  "/umd/index-browser-experimental-umd.js",
  "(function (global, factory) {
  if (typeof define === \\"function\\" && define.amd) {
    define([\\"exports\\"], factory);
  } else if (typeof exports !== \\"undefined\\") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.examples = mod.exports;
  }
})(typeof globalThis !== \\"undefined\\" ? globalThis : typeof self !== \\"undefined\\" ? self : this, function (_exports) {
  \\"use strict\\";

  Object.defineProperty(_exports, \\"__esModule\\", {
    value: true
  });
  _exports.run = run;
  _exports.wait = wait;

  // Generated with Packemon: https://packemon.dev
  // Platform: browser, Support: experimental, Format: umd
  async function wait() {
    return new Promise(resolve => {
      setTimeout(resolve, 100);
    });
  }

  async function run() {
    try {
      await wait();
    } catch {
      throw new Error('Fail');
    }
  }
});
//# sourceMappingURL=index-browser-experimental-umd.js.map
",
]
`;

exports[`Async/await transforms example test case 28`] = `
Array [
  "/umd/index-browser-legacy-umd.js",
  "function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

(function (global, factory) {
  if (typeof define === \\"function\\" && define.amd) {
    define([\\"exports\\"], factory);
  } else if (typeof exports !== \\"undefined\\") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.examples = mod.exports;
  }
})(typeof globalThis !== \\"undefined\\" ? globalThis : typeof self !== \\"undefined\\" ? self : this, function (_exports) {
  \\"use strict\\";

  Object.defineProperty(_exports, \\"__esModule\\", {
    value: true
  });
  _exports.wait = _exports.run = void 0;

  function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

  var run = function run() {
    try {
      return Promise.resolve(_catch(function () {
        return Promise.resolve(wait()).then(function () {});
      }, function () {
        throw new Error('Fail');
      }));
    } catch (e) {
      return Promise.reject(e);
    }
  };

  _exports.run = run;

  // Generated with Packemon: https://packemon.dev
  // Platform: browser, Support: legacy, Format: umd
  var wait = function wait() {
    var _this = this;

    try {
      return Promise.resolve(new Promise(function (resolve) {
        _newArrowCheck(this, _this);

        setTimeout(resolve, 100);
      }.bind(this)));
    } catch (e) {
      return Promise.reject(e);
    }
  };

  _exports.wait = wait;
});
//# sourceMappingURL=index-browser-legacy-umd.js.map
",
]
`;

exports[`Async/await transforms example test case 29`] = `
Array [
  "/umd/index-browser-stable-umd.js",
  "function _catch(body, recover) {
  try {
    var result = body();
  } catch (e) {
    return recover(e);
  }

  if (result && result.then) {
    return result.then(void 0, recover);
  }

  return result;
}

(function (global, factory) {
  if (typeof define === \\"function\\" && define.amd) {
    define([\\"exports\\"], factory);
  } else if (typeof exports !== \\"undefined\\") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.examples = mod.exports;
  }
})(typeof globalThis !== \\"undefined\\" ? globalThis : typeof self !== \\"undefined\\" ? self : this, function (_exports) {
  \\"use strict\\";

  Object.defineProperty(_exports, \\"__esModule\\", {
    value: true
  });
  _exports.wait = _exports.run = void 0;

  function _newArrowCheck(innerThis, boundThis) { if (innerThis !== boundThis) { throw new TypeError(\\"Cannot instantiate an arrow function\\"); } }

  var run = function run() {
    try {
      return Promise.resolve(_catch(function () {
        return Promise.resolve(wait()).then(function () {});
      }, function () {
        throw new Error('Fail');
      }));
    } catch (e) {
      return Promise.reject(e);
    }
  };

  _exports.run = run;

  // Generated with Packemon: https://packemon.dev
  // Platform: browser, Support: stable, Format: umd
  var wait = function wait() {
    var _this = this;

    try {
      return Promise.resolve(new Promise(function (resolve) {
        _newArrowCheck(this, _this);

        setTimeout(resolve, 100);
      }.bind(this)));
    } catch (e) {
      return Promise.reject(e);
    }
  };

  _exports.wait = wait;
});
//# sourceMappingURL=index-browser-stable-umd.js.map
",
]
`;
